openedx/core/djangoapps/catalog/tests/test_utils.py

```diff
@@ -1,37 +1,41 @@
 """Tests covering utilities for integrating with the catalog service."""
+# pylint: disable=missing-docstring
 import uuid
 import copy
 
+from django.contrib.auth import get_user_model
 from django.test import TestCase
 import mock
 from opaque_keys.edx.keys import CourseKey
 
-from openedx.core.djangoapps.catalog import utils
 from openedx.core.djangoapps.catalog.models import CatalogIntegration
-from openedx.core.djangoapps.catalog.tests import factories, mixins
-from student.tests.factories import UserFactory, AnonymousUserFactory
+from openedx.core.djangoapps.catalog.tests.factories import ProgramFactory, ProgramTypeFactory
+from openedx.core.djangoapps.catalog.tests.mixins import CatalogIntegrationMixin
+from openedx.core.djangoapps.catalog.utils import get_programs, munge_catalog_program, get_program_types, get_programs_with_type_logo
+from student.tests.factories import UserFactory
+
 
 UTILS_MODULE = 'openedx.core.djangoapps.catalog.utils'
+User = get_user_model()
 
 
 @mock.patch(UTILS_MODULE + '.get_edx_api_data')
-# ConfigurationModels use the cache. Make every cache get a miss.
-@mock.patch('config_models.models.cache.get', return_value=None)
-class TestGetPrograms(mixins.CatalogIntegrationMixin, TestCase):
+class TestGetPrograms(CatalogIntegrationMixin, TestCase):
     """Tests covering retrieval of programs from the catalog service."""
     def setUp(self):
         super(TestGetPrograms, self).setUp()
 
-        self.user = UserFactory()
         self.uuid = str(uuid.uuid4())
         self.type = 'FooBar'
         self.catalog_integration = self.create_catalog_integration(cache_ttl=1)
 
+        UserFactory(username=self.catalog_integration.service_username)
```

Comment:
Yes and no. We need to user saved to the db so that code under test will function as desired; that's what this is doing. However, we don't need to keep a reference to it around in this test (i.e., no need to assign the return value to a variable).